pdns/pdns_recursor.cc

```diff
@@ -2002,7 +2008,11 @@ static void handleRunningTCPQuestion(int fd, FDMultiplexer::funcparam_t& var)
     }
     conn->bytesread+=(uint16_t)bytes;
     if(conn->bytesread==conn->qlen) {
-      t_fdm->removeReadFD(fd); // should no longer awake ourselves when there is data to read
+      conn->d_requestsInFlight++;
+      if (conn->d_requestsInFlight >= TCPConnection::s_maxInFlight) {
+        //cerr << "Disabling... " << conn->d_requestsInFlight << ' ' << fd << endl;
+        t_fdm->removeReadFD(fd); // should no longer awake ourselves when there is data to read
+      }
```

Comment:
Shouldn't we update the read TTD when we don't disable the polling?

Reply:
yes, I think so. The original code didn't do that, but it should have.